sms_sponsorship/forms/sms_sponsorship_registration_form.py

```diff
@@ -186,11 +188,22 @@ def form_before_create_or_update(self, values, extra_values):
                     })
             sponsorship.on_change_group_id()
 
-        def form_after_create_or_update(self, values, extra_values):
+        @job(default_channel="root.res_partner")
+        @related_action(action='related_action_update_partner')
+        def update_partner(self, partner_vals):
+            self.env['res.partner'].search([
+                ('name', '=', self.partner_name),
+                ('email', '=', self.partner_email)
+            ]).write(partner_vals)
+
+        @job(default_channel="root.recurring_contract")
+        @related_action(action='related_action_finalize_form')
+        def finalize_form(self, sponsorship):
             # validate sponsorship and send confirmation email
-            sponsorship = self.main_object.sudo()
+            # we use partner name temporarily for testing reasons, should
+            # use sponsorship.id
             sms_request = self.env['sms.child.request'].sudo().search([
-                ('sponsorship_id', '=', self.main_object.id)
+                ('partner_id.name', '=', sponsorship.partner_id.name)
```

Comment:
No, the previous search was actually better.

Reply:
I know, it was just not working on tests because I can't create a sms_request with an attributed sponsorship so I couldn't find a sms_request.